includes/Checker/Preparations/Force_Single_Plugin_Preparation.php

```diff
@@ -74,39 +75,67 @@ public function prepare() {
 	}
 
 	/**
-	 * Filter active plugins.
+	 * Filters active plugins to only include required ones.
+	 *
+	 * This means:
+	 *
+	 * * The plugin being tested
+	 * * All dependencies of the plugin being tested
+	 * * Plugin Check itself
+	 * * All plugins depending on Plugin Check (they could be adding new checks)
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param array $active_plugins List of active plugins.
-	 * @return array List of active plugins.
+	 * @param mixed $active_plugins List of active plugins.
+	 * @return mixed List of active plugins.
 	 */
 	public function filter_active_plugins( $active_plugins ) {
-		if ( is_array( $active_plugins ) && in_array( $this->plugin_basename, $active_plugins, true ) ) {
+		if ( ! is_array( $active_plugins ) ) {
+			return $active_plugins;
+		}
 
-			if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
-				$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
-			} else {
-				$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
-				$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
-			}
+		// The plugin being tested isn't actually active yet.
+		if ( ! in_array( $this->plugin_basename, $active_plugins, true ) ) {
+			return $active_plugins;
+		}
 
-			$plugin_base_file = plugin_basename( $plugin_check_file );
+		if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
+			$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
+		} else {
+			$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
+			$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
+		}
 
-			// If the plugin-check is the only available plugin then return that one only.
-			if ( $this->plugin_basename === $plugin_base_file ) {
+		$plugin_check_basename = plugin_basename( $plugin_check_file );
 
-				return array(
-					$plugin_base_file,
-				);
-			}
+		$new_active_plugins = array(
+			$this->plugin_basename, // Plugin to test.
+			$plugin_check_basename, // Plugin Check itself.
+		);
 
-			return array(
-				$this->plugin_basename,
-				$plugin_base_file,
+		// Plugin dependencies support was added in WordPress 6.5.
+		if ( class_exists( 'WP_Plugin_Dependencies' ) ) {
+			WP_Plugin_Dependencies::initialize();
+
+			$new_active_plugins = array_merge(
+				$new_active_plugins,
+				WP_Plugin_Dependencies::get_dependencies( $this->plugin_basename )
+			);
+
+			$new_active_plugins = array_merge(
+				$new_active_plugins,
+				// Include any dependents of Plugin Check, but only if they were already active.
+				array_filter(
+					WP_Plugin_Dependencies::get_dependents( dirname( $plugin_check_basename ) ),
```

Comment:
What a confusing API, requiring a plugin _file_ for `get_dependencies()` but a plugin _slug_ for `get_dependents()`... 🙄